### PR TITLE
add TryGetPrimaryDepartment to jobs system

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
add notion of primary departments

a primary department is the actual physical department a job belongs to, but excluding hop and command since they dont have one.

engi med sec etc are primary departments, command and station specific are not.

## Why / Balance
for turf war

## Technical details
unit test is idk

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by chromiumboy** [comment]
Technically HoP is the head of the service department. Captain is the only role that solely belongs to Command

**Comment by deltanedas** [comment]
hop going around spraying everything as maints seems weird though, and hop is second in command

**Comment by Emisse** [comment]
wqhat do you envision this being used for

**Comment by GoodWheatley** [comment]
> wqhat do you envision this being used for

It’s for turf war

**Comment by deltanedas** [comment]
> Why / Balance

:trollface:

File: Content.Shared/Roles/Jobs/SharedJobSystem.cs [hunk 1]
```diff
@@ -76,6 +76,29 @@ public bool TryGetDepartment(string jobProto, [NotNullWhen(true)] out Department
...
 
+    /// <summary>
+    /// Like <see cref="TryGetDepartment"/> but ignores any non-primary departments.
+    /// For example, with CE it will return Engineering but with captain it will
+    /// not return anything, since Command is not a primary department.
+    /// </summary>
+    public bool TryGetPrimaryDepartment(string jobProto, [NotNullWhen(true)] out DepartmentPrototype? departmentPrototype)
+    {
+        // not sorting it since there should only be 1 primary department for a job.
+        var departmentProtos = _protoManager.EnumeratePrototypes<DepartmentPrototype>().ToList();
```
Comment (metalgearsloth) [inline Content.Shared/Roles/Jobs/SharedJobSystem.cs:87]: Why do you need the ToList allocation

Comment (deltanedas) [reply Content.Shared/Roles/Jobs/SharedJobSystem.cs:87]: i think IEnumerable can only be iterated once then its empty

Comment (mirrorcult) [reply Content.Shared/Roles/Jobs/SharedJobSystem.cs:87]: youre only iterating it once though

Comment (deltanedas) [reply Content.Shared/Roles/Jobs/SharedJobSystem.cs:87]: oh i thought this was a comment on the test :trollface:

**Pull request merged**